Author: Upstream
Description: Patch fixes segfault for matching exact string with
 non-ASCII characters in it.
Bug: https://sourceforge.net/tracker/?func=detail&aid=3095935&group_id=13179&atid=313179

--- expect-5.44.1.15.orig/expect.c
+++ expect-5.44.1.15/expect.c
@@ -946,6 +946,9 @@
 	expDiagLogU(expPrintify(Tcl_GetString(e->pat)));
 	expDiagLog("\"? ");
 	if (p) {
+	    /* Bug 3095935. Go from #bytes to #chars */
+	    patLength = Tcl_NumUtfChars (pat, patLength);
+
 	    e->simple_start = p - str;
 	    o->e = e;
 	    o->matchlen = patLength;
